client/landing/stepper/declarative-flow/internals/index.tsx

```diff
@@ -216,6 +226,11 @@ export const FlowRenderer: React.FC< { flow: Flow; steps: readonly StepperStep[]
 		<Loading className="wpcom-loading__boot" />
 	);
 
+	let stepSlugToRedirectTo = flow.__experimentalUseBuiltinAuth ? firstStepSlug : stepPaths[ 0 ];
+	if ( isLoggedIn && customInitialStepSlug ) {
+		stepSlugToRedirectTo = customInitialStepSlug;
```

Comment:
If this user is logged out, the hook will be ignored forever. Even after they log in. No?

Reply:
Good catch on this one. I've now changed the logic here in two ways:
 * When `customInitialStepSlug` is defined, we use that instead of these values.
 * I've also added this to the possible values for `postAuthStepSlug` in the logic that identifies what flow to redirect to after logging in
   - I am less sure about this logic, as I think that this should already be handled by the auth logic that detects the user is logged out and redirects for that case